azure-core/azure/core/credentials_async.py

```diff
@@ -0,0 +1,18 @@
+# -------------------------------------------------------------------------
```

Comment:
Do we really need this file/class?

Reply:
I'm not in love with it but getting a token often requires a network request. Is there a better way to accommodate sync and async here?

Reply:
This specific class is just an ABC. And from what I can tell, it is only used to help with type checking for `AsyncBearerTokenCredentialPolicy.__init__` (the only difference between this class and the sync `TokenCredentials` class is the `async def` vs `def` of the `get_token` method.

Reply:
Correct, I created it only so the async policy can await `get_token`. I assumed the decorator would require subclasses to define `get_token` as a coroutine; today I learned this is not the case 👨‍🎓